base/operators.jl

```diff
@@ -293,7 +293,7 @@ fldmod1{T<:Real}(x::T, y::T) = (fld1(x,y), mod1(x,y))
 fldmod1{T<:Integer}(x::T, y::T) = (fld1(x,y), mod1(x,y))
 
 # transpose
-transpose(x) = x
+transpose(x) = error("transpose not implemented for $(typeof(x)). Consider permutedims.")
```

Comment:
No reason beyond mimicking https://github.com/JuliaLang/julia/blob/9a66205d071bd013b5c0c607764a0885dd4e38f3/base/abstractarraymath.jl#L9-L10. Should I change it to `ArgumentError`, and if so should I change the linked definitions to `ArgumentError`s as well? Thanks for reviewing! :)